### PR TITLE
Feature/sync committees on app dashboard (BIDS-1153)

### PR DESCRIPTION
**Pull request by D13ce**

### CODE REVIEW

File: handlers/api.go [hunk 1]
```diff
@@ -839,6 +848,177 @@ func getSyncCommitteeFor(validators []uint64, period uint64) ([]interface{}, err
...
+		uniquePeriods[utils.SyncPeriodOfEpoch(firstSyncEpoch)] = true
+
+		// exit epoch (if any)
+		if v.ExitEpoch != noExitEpoch && v.ExitEpoch > firstSyncEpoch {
+			uniquePeriods[utils.SyncPeriodOfEpoch(v.ExitEpoch)] = true
+		}
+	}
+
+	// transform map to slice; this will be used to query sync_committees_count_per_validator
+	timeStampSlice := make([]uint64, 0, len(uniquePeriods))
```
Comment (manuelsc) [inline handlers/api.go:902]: Isn't this a period slice not timestamp?

File: handlers/api.go [hunk 2]
```diff
@@ -839,6 +848,177 @@ func getSyncCommitteeFor(validators []uint64, period uint64) ([]interface{}, err
...
+	if err != nil {
+		return nil, err
+	}
+
+	r.ParticipatedSlots = uint64(syncStats.Participated)
+	r.MissedSlots = uint64(syncStats.Missed)
+
+	// validator_stats is updated only once a day, everything missing has to be collected from bigtable (which is slower than validator_stats)
+	// check when the last update to validator_stats was
+	var lastExportedDay uint64
```
Comment (manuelsc) [inline handlers/api.go:972]: Can we move the loading of fresh stats from bigtable to it's own function? This function is getting me too big

**Review by manuelsc — CHANGES_REQUESTED** [review]

File: exporter/sync_committees_count.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+			firstPeriod = dbPeriod + 1
+		}
+
+		err = db.WriterDb.Get(&countSoFar, `SELECT count_so_far FROM sync_committees_count_per_validator WHERE period = $1`, dbPeriod)
+		if err != nil {
+			return err
+		}
+	}
+
+	for p := firstPeriod; p <= currentPeriod; p++ {
```
[the remark this answers is not in the document]
Comment (D13ce) [reply exporter/sync_committees_count.go:52]: Changed.

File: handlers/api.go (hunk 1 above)
Comment (D13ce) [reply handlers/api.go:902]: You are right, changed.

File: handlers/api.go (hunk 2 above)
Comment (D13ce) [reply handlers/api.go:972]: Sure can do, changed.

**Comment by D13ce** [comment]
@manuelsc, during refactoring I have found a bug related to Altair:

`required period not found` was reported as `periodInfoMap` whenever validators were checked that were activated before Altair. 
Fixed via `FirstPossibleSyncCommittee` (see [HERE](https://github.com/gobitfly/eth2-beaconchain-explorer/pull/1743/files#diff-6a59605178bab56012cab99c23c5c9de613ff8ecc0af6a1baf5d010c757dda31R885)).

File: handlers/api.go [hunk 4]
```diff
@@ -839,6 +848,194 @@ func getSyncCommitteeFor(validators []uint64, period uint64) ([]interface{}, err
...
+
+		// exit epoch (if any)
+		if validatorsInfo[i].ExitEpoch != noExitEpoch && validatorsInfo[i].ExitEpoch > firstSyncEpoch {
+			uniquePeriods[utils.SyncPeriodOfEpoch(validatorsInfo[i].ExitEpoch)] = true
+		}
+	}
+
+	// transform map to slice; this will be used to query sync_committees_count_per_validator
+	periodSlice := make([]uint64, 0, len(uniquePeriods))
+	for ts := range uniquePeriods {
```
Comment (manuelsc) [inline handlers/api.go:919]: minor readability detail, period instead of ts

**Review by manuelsc — APPROVED** [review]
lgtm

**Pull request merged**